pandas/tests/test_index.py

```diff
@@ -5673,6 +5673,13 @@ def test_equals_operator(self):
         # GH9785
         self.assertTrue((self.index == self.index).all())
 
+    def test_subclassing(self):
+        # GH11267
+        class MyMultiIndex(MultiIndex):
+            pass
+        mi = MyMultiIndex([['a'], ['b']], [[0], [0]])
+        self.assertTrue(isinstance(mi, MyMultiIndex))
```

Comment:
this needs a fair bit of testing, e.g. run thru the some typical operations of a MI, just with a sub-class one.